### PR TITLE
update encoder.py for 2018 some more

### PR DESCRIPTION
**Pull request by ariovistus**

### CODE REVIEW

**Comment by auscompgeek** [comment]
Heh, nice. `getEncoderFPGAIndex` is unimplemented in the simulated HAL.

**Comment by auscompgeek** [comment]
@Twinters007 implemented it in #388 for you. Rerunning the tests now.

**Comment by auscompgeek** [comment]
Uh, right, rerunning a build doesn't pull in changes to the build config.

@ariovistus Could you please rebase this PR?

**Comment by Twinters007** [comment]
That's lame

File: wpilib/wpilib/encoder.py [hunk 1]
```diff
@@ -409,9 +419,9 @@ def setIndexSource(self, source, indexing_type=IndexingType.kResetOnRisingEdge):
         Set the index source for the encoder. When this source rises, the encoder count automatically resets.
 
         :param source: Either an initialized DigitalSource or a DIO channel number
-        :type: Either a :class:`wpilib.DigitalInput` or number
+        :type: Either a :class:`.DigitalInput` or number
```
Comment (Twinters007) [inline wpilib/wpilib/encoder.py]: Does this work if you dont say `:type source:`

Comment (Twinters007) [inline wpilib/wpilib/encoder.py]: You need to say `:type source:`

File: wpilib/wpilib/encoder.py [hunk 2]
```diff
@@ -409,9 +419,9 @@ def setIndexSource(self, source, indexing_type=IndexingType.kResetOnRisingEdge):
...
         :param indexing_type: The state that will cause the encoder to reset
-        :type: A value from :class:`wpilib.DigitalInput.IndexingType`
+        :type: A value from :class:`.IndexingType`
```
Comment (Twinters007) [inline wpilib/wpilib/encoder.py]: Same thing here

**Comment by ariovistus** [comment]
I made IndexingType an IntEnum, which makes setIndexSource's docs show up like

setIndexSource(source, indexing_type=<IndexingType.kResetOnRisingEdge: 3>)

which I think is an improvement, but it looks a little weird

**Comment by Twinters007** [comment]
I think that's ok. Better than just saying 3.

**Pull request merged**